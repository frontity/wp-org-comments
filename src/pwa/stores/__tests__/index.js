import Comments from '..';
import { types } from 'mobx-state-tree';
import withComments from './data/withComments.json';
import withoutComments from './data/withoutComments.json';
import comments from './data/comments.json';

const Connection = types.model('Connection', {}).views(() => ({
  entity: (type, id) => (id === 60 ? withComments : withoutComments),
}));

const Stores = types.model('Stores', {
  comments: types.optional(Comments, {}),
  connection: types.optional(Connection, {}),
});

let stores;
let request;
beforeEach(() => {
  request = {
    post: jest.fn(() => request),
    type: jest.fn(() => request),
    get: jest.fn(() => Promise.resolve({ body: comments })),
    send: jest.fn(() => Promise.resolve()),
  };
  stores = Stores.create({}, { request });
});

describe('wp-org-comments â€º Comments', () => {
  test('fromPost returns an empty array for posts without comments', () => {
    const rootComments = stores.comments.fromPost({ type: 'post', id: 56 });
    expect(rootComments).toMatchSnapshot();
    expect(request.get.mock.calls).toHaveLength(0);
  });
  test('fromPost returns comments from entity.raw', () => {
    const rootComments = stores.comments.fromPost({ type: 'post', id: 60 });
    expect(rootComments).toMatchSnapshot();
    rootComments.forEach(comment => {
      expect(comment.replies).toMatchSnapshot();
    });
  });
  test('update comments', async () => {
    await stores.comments.update({ type: 'post', id: 60 });
    expect(stores.comments).toMatchSnapshot();
  });
  test('submit a new comment', async () => {
    const data = {
      comment: 'Hello',
      author: 'Mario',
      email: 'mario@frontity.com',
      url: 'http://frontity.com',
      comment_post_ID: 60,
      comment_parent: 129,
    };
    await stores.comments.submit({ type: 'post', id: 60, data });
    expect(stores.comments).toMatchSnapshot();
    expect(request.post).toHaveBeenCalledWith('/wp-comments-post.php');
    expect(request.type).toHaveBeenCalledWith('form');
    expect(request.send).toHaveBeenCalledWith(data);
  });
  test('sort replies by date', async () => {
    await stores.comments.update({ type: 'post', id: 60 });
    const rootComments = stores.comments.fromPost({ type: 'post', id: 60 });
    expect(rootComments[1].replies).toMatchSnapshot();
  });
});
