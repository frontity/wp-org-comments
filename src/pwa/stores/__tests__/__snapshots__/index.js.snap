// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wp-org-comments â€º Comments fromPost returns an empty array for posts without comments 1`] = `Array []`;

exports[`wp-org-comments â€º Comments fromPost returns comments from entity.raw 1`] = `
Array [
  Object {
    "avatar": "https://secure.gravatar.com/avatar/485a5c16c5e51ffad6e76c48fd0cb88b?s=96&d=mm&r=g",
    "content": "<p>Nice post, Alan! Iceland must be amazing. Thanks for sharing ðŸ™‚</p>
",
    "dateString": "2017-01-13T16:10:43",
    "id": "5",
    "name": "Reyes Martinez",
    "parent": "0",
  },
  Object {
    "avatar": "https://secure.gravatar.com/avatar/a850f5724ad7ea03941081972841a04a?s=96&d=mm&r=g",
    "content": "<p>Hi, Reyes</p>
",
    "dateString": "2019-02-06T17:35:39",
    "id": "124",
    "name": "Mario",
    "parent": "0",
  },
]
`;

exports[`wp-org-comments â€º Comments fromPost returns comments from entity.raw 2`] = `Array []`;

exports[`wp-org-comments â€º Comments fromPost returns comments from entity.raw 3`] = `
Array [
  Object {
    "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
    "content": "<p>Hi, Mario</p>
",
    "dateString": "2019-02-08T17:28:33",
    "id": "129",
    "name": "David Arenas",
    "parent": "124",
  },
]
`;

exports[`wp-org-comments â€º Comments sort replies by date 1`] = `
Array [
  Object {
    "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
    "content": "<p>Hi, Mario</p>
",
    "dateString": "2019-02-08T17:28:33",
    "id": "129",
    "name": "David Arenas",
    "parent": "124",
  },
  Object {
    "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
    "content": "<p>Hi, Mario</p>
",
    "dateString": "2019-02-08T18:00:00",
    "id": "131",
    "name": "Alan",
    "parent": "124",
  },
]
`;

exports[`wp-org-comments â€º Comments submit a new comment 1`] = `
Object {
  "commentsMap": Object {
    "60": Object {
      "124": Object {
        "avatar": "https://secure.gravatar.com/avatar/a850f5724ad7ea03941081972841a04a?s=96&d=mm&r=g",
        "content": "<p>Hi, Reyes</p>
",
        "dateString": "2019-02-06T17:35:39",
        "id": "124",
        "name": "Mario",
        "parent": "0",
      },
      "129": Object {
        "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
        "content": "<p>Hi, Mario</p>
",
        "dateString": "2019-02-08T17:28:33",
        "id": "129",
        "name": "David Arenas",
        "parent": "124",
      },
      "130": Object {
        "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
        "content": "<p>Hello</p>
",
        "dateString": "2019-02-08T17:28:40",
        "id": "130",
        "name": "Mario",
        "parent": "129",
      },
      "131": Object {
        "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
        "content": "<p>Hi, Mario</p>
",
        "dateString": "2019-02-08T18:00:00",
        "id": "131",
        "name": "Alan",
        "parent": "124",
      },
      "5": Object {
        "avatar": "https://secure.gravatar.com/avatar/485a5c16c5e51ffad6e76c48fd0cb88b?s=96&d=mm&r=g",
        "content": "<p>Nice post, Alan! Iceland must be amazing. Thanks for sharing ðŸ™‚</p>
",
        "dateString": "2017-01-13T16:10:43",
        "id": "5",
        "name": "Reyes Martinez",
        "parent": "0",
      },
    },
  },
}
`;

exports[`wp-org-comments â€º Comments update comments 1`] = `
Object {
  "commentsMap": Object {
    "60": Object {
      "124": Object {
        "avatar": "https://secure.gravatar.com/avatar/a850f5724ad7ea03941081972841a04a?s=96&d=mm&r=g",
        "content": "<p>Hi, Reyes</p>
",
        "dateString": "2019-02-06T17:35:39",
        "id": "124",
        "name": "Mario",
        "parent": "0",
      },
      "129": Object {
        "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
        "content": "<p>Hi, Mario</p>
",
        "dateString": "2019-02-08T17:28:33",
        "id": "129",
        "name": "David Arenas",
        "parent": "124",
      },
      "130": Object {
        "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
        "content": "<p>Hello</p>
",
        "dateString": "2019-02-08T17:28:40",
        "id": "130",
        "name": "Mario",
        "parent": "129",
      },
      "131": Object {
        "avatar": "https://secure.gravatar.com/avatar/0e185711dda10636f58cf83e358b2e6e?s=96&d=mm&r=g",
        "content": "<p>Hi, Mario</p>
",
        "dateString": "2019-02-08T18:00:00",
        "id": "131",
        "name": "Alan",
        "parent": "124",
      },
      "5": Object {
        "avatar": "https://secure.gravatar.com/avatar/485a5c16c5e51ffad6e76c48fd0cb88b?s=96&d=mm&r=g",
        "content": "<p>Nice post, Alan! Iceland must be amazing. Thanks for sharing ðŸ™‚</p>
",
        "dateString": "2017-01-13T16:10:43",
        "id": "5",
        "name": "Reyes Martinez",
        "parent": "0",
      },
    },
  },
}
`;
